### ISSUE TITLE
env-loginInteractiveShell.json files appear

### ISSUE THREAD

**Issue by markedwards** [microsoft/vscode-remote-release#7962]
- VSCode Version: 1.75.0
- Local OS Version: MacOS 13.2
- Remote OS Version: Debian bullseye
- Remote Extension/Connection Type: Containers/WSL/Server

- Logs:

Steps to Reproduce:

1. Start a remote container VSCode window
2. See a `tmp/.../env-loginInteractiveShell.json` file created in the source tree

What is this file, and why is it injected into my project source?

**Comment by chrmarti** [comment]
This is a cache file and shouldn't end up in your workspace folder. How did you create the dev container? What is the full path of that file inside the container?

**Comment by daulis** [comment]
@chrmarti I also have the same issue.

Path: /workspaces/PROJECT_NAME/tmp/devcontainers-LONG_GUID/env-loginInteractiveShell.json

For me:
1. `cd PROJECT && code .` (In WSL, but probably not relevant)
2. `Dev Containers: Reopen in Container`
3. File created

A new file is created each time I open the container

**Comment by markedwards** [comment]
My dev container is attached to a remote docker container.  For me the path is:
```
/cargo/api/tmp/devcontainers-LONG_GUID/env-loginInteractiveShell.json
```

For me, I'm creating the dev container by choosing "Attach Visual Studio Code" in the Docker extension.

**Comment by trxcllnt** [comment]
Also seeing this. It seems to come from this line in the logs:
```shell
[5809 ms] Start: Run in container: mkdir -p 'tmp/devcontainers-18936752-e8db-449a-ac57-6af6de14814a1675467355486' && cat > 'tmp/devcontainers-18936752-e8db-449a-ac57-6af6de14814a1675467355486/env-loginInteractiveShell.json' << 'envJSON'
```

My container's `WORKDIR` is configured to be my project's repo root, so making a dir like this will put it in the source tree.

**Comment by chrmarti** [comment]
Steps to reproduce / verify:

Reopen a folder with the following `.devcontainer/devcontainer.json` in a container:
```json
{
	"image": "mcr.microsoft.com/devcontainers/base:latest",
	"runArgs": [
		"-w", "${containerWorkspaceFolder}"
	]
}
```

This will add a `tmp` folder to the workspace folder with the mentioned subfolder and JSON file.

Verify by ensuring this no longer happens.

**Comment by chrmarti** [comment]
Confirmed `/tmp` should always be available: https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch03s18.html.

**Comment by aeschli** [comment]
Verified that the [example](https://github.com/microsoft/vscode-remote-release/issues/7962#issuecomment-1418739209) creates `/tmp/devcontainers-d92a34d3-6ee8-475b-96c5-5e279770c8031675694306663`

**Comment by chrmarti** [comment]
Fixed in Dev Containers 0.275.1 and 0.277.0-pre-release.

**Comment by markedwards** [comment]
I can confirm it's fixed for me now.  Magic!

Thanks.
